### ISSUE TITLE
Only one instance of Pokebuddy works at a time

### ISSUE THREAD

**Issue by Sorata5000** [finndev/PokeBuddy#669]
So I found an interesting bug. Not sure what update did it but once I start a second instance of pokebuddy while playing on Nox the Nox servers stop working (i.e. cant click on gyms or pokestops, once relogged in gyms and pokestops dont show up on the maps). The 2nd instance of Pokebuddy works for a few minutes then just stops hitting pokestops and finding pokemon. Shortly after the 2nd instance stops working the first instance starts doing the same thing. Before I could run up to 3 instances of Pokebuddy and Nox at the same time with zero issue. Iv already deleted the folders and re installed numerous times and the same thing happens. Also, I am able to recreate the bug.

**Comment by curtiola** [comment]
2 accounts signed in on the same machine results in a soft ban, it's obvious at that point to Niantic that it's not a cell phone, they added this in 0.33.0

**Comment by Sorata5000** [comment]
I'm not logged into the same account. The account Im using on Nox are not being used on any instance of Pokebuddy.

**Comment by curtiola** [comment]
No, I said 2 accounts on the same machine, 2 people using the same IP is impossible on a cell phone yo

**Comment by Sorata5000** [comment]
I don't understand then. It was working fine last night with 3 accounts logged in on Pokebuddy. Also, it works perfectly when I have one instance of Pokebuddy open and playing Nox at the same time.

**Comment by curtiola** [comment]
Niantic updated to 0.33.0 yesterday, now no more than 1 account can be signed in per IP

**Comment by curtiola** [comment]
Does it work with no and pokebuddy at the same time today? And for an extended period of time?

**Comment by Dropholion** [comment]
Yeah I've had this too, I guess it's limited now.

**Comment by Sorata5000** [comment]
yes Curtiola. Before I started troubleshooting the bug, I had a Pokebuddy account farming and was using Nox with 2 completely separate accounts and it worked for 30 minutes with zero issue. Then tested opening a 2nd instance of Pokebuddy with a completely separate bot account I have. The 2nd instance of Pokebuddy worked for a few minutes, then stopped clicking on pokestops. Its not the same as a softban because even when the account is softbanned, pokebuddy will still attempt to click stops giving 0 xp earned messages. It now just stops clicking the pokestop all together, then the first instance of Pokebuddy does the same thing. And around the same time the account I use while playing on Nox stops working. And by not working I mean every pokestop shows "Please try again later" and gyms just become unclickable. Then relogging on Nox while the pokebuddys are open makes Nox show zero pokestops or gyms on the game in game.

**Comment by dallashowell89** [comment]
I am experiencing the same issue.

**Comment by Sorata5000** [comment]
Vendunk, I'm not having that issue. I have 1 instance of Pokebuddy that works with zero issue. Pokestops will still show up after the 2nd instance of pokebuddy is opened. I am not having that issue.

**Comment by Sorata5000** [comment]
I will try again to see if pokestops disappear with only one instance of pokebuddy open. Will report back. I will re-attempt to open a 2nd instance after 45 min pass.

**Comment by curtiola** [comment]
Oh OK, interesting, but how long does having nox and one instance of pokebuddy running last for? Others are reporting after about 30 minutes or so with 2 instances of pokebuddy open they have to restart pokebuddy

**Comment by Sorata5000** [comment]
I'm going to test for 45 min to an hour with one instance of pokebuddy open and one instance of nox open and see what happens. 5 mins in and zero issue.

**Comment by zeek11212** [comment]
![image](https://cloud.githubusercontent.com/assets/20785899/17537762/de05b598-5e64-11e6-8e3a-65998a9e466c.png)

2 accounts work fine for me

**Comment by curtiola** [comment]
I'm curious, but either way, running more than 1 instance from the same IP is basically a guaranteed red flag on all the connected accounts and I wouldn't recommend it even if it does still work somehow

**Comment by curtiola** [comment]
Perhaps I was mistaken about the update then, sorry if I was, but still it's not a good idea

**Comment by Sorata5000** [comment]
There only bot accounts. Really don't mind if they get banned lol. Main reason I'm doing more than one is if I have any friends with kids who really need an account I can pass it to them. And its all good Curtiola, no need to be sorry. :)

**Comment by Sorata5000** [comment]
30 mins in and zero issues on Nox and Pokebuddy.

**Comment by curtiola** [comment]
Well that's good, also strange, if multiples are allowed still I have no clue why another pokebuddy would cause an issue lol

**Comment by psblackz** [comment]
just use 2 bots.. more than i think will cause problems, like errors, and dont farm anything

**Comment by Sorata5000** [comment]
Same thing happend. Almost an hour with one instance of Nox open and one instance of Pokebuddy open. Once the 2nd instance of Pokebuddy starts, a few minutes after the 2nd instance of pokebuddy stops, nox stops responding to the servers but the 1st instance of poke buddy seems to work.

**Comment by Sorata5000** [comment]
Ok. Now this is odd. Tried this again. First instance of Pokebuddy has Catch Pokemon unchecked but farm, transfer and recycle are checked. The 2nd instance of pokebuddy has catch pokemon on and its working perfectly now.

**Comment by IamDeric** [comment]
hmmnn... so i try this for 5x already. I use my usual botting account (3 account in the same time.) yes after 20min of using it all it will automatically stop. even with only 2 account in the same time, it will stop moving and just stay in the same spot. so i thought at first its only a bug or an issue to be fix with the bot, but its not, as mention above its the IP. now im using just 1 account no problem running it for 15mins now.and its still good.

**Comment by bt** [comment]
I used to run 6 bots on 1 machine and that was fine.

I don't understand how they can `red flag` you if you have multiple accounts on the same IP... My 2 brothers, my sister and my mum could all be playing at the same time on our Wifi connection...

**Comment by curtiola** [comment]
Maybe it's the Mac address then lol they can probably see what devices are connected to their network like you can see them on your router's settings page, idk.. maybe it's nothing to do with multiple accounts on the same device

**Comment by curtiola** [comment]
These type of problems just started arising with the 0.33.0 update is all I know

**Comment by bt** [comment]
From what I've seen in the protocol design, they don't send MAC addresses to the server. And because MAC address is on a different layer, they wouldn't be able to see it through the communication.

**Comment by curtiola** [comment]
Good to know, maybe finndev did something with his pokebuddy update yesterday that caused it

**Comment by Thorben24** [comment]
@curtiola @bt @Sorata5000 

Its an IP ban I think.

If i run PokeBuddy 2 times with 2 different accounts on the same computer with the same IP adress, after 20 minutes, the account isnt on a position to catch pokemons / farm pokestops anymore. Via Bluestacks i logged in and saw, that there are no more pokestops on the map, no pokemons spawning. I cant activate items etc. (I checked it 3 times, it wasnt a bug).

After I restarted my FritzBox (Ip Adress Reset) everything worked. I didnt get a softban. 

Right now I run Pokebuddy 2 times on the same machine, but I filter 1 instance via a proxy server. Since 5 hours everything is working without a single problem yet.

**Comment by curtiola** [comment]
Hm, I'm not sure, @bt has a good point, my brother and I play at the same time on the same WiFi connection with our phones and we don't get banned, before I was thinking local IP, not gateway (which was a bit ignorant of me lol), but they don't see local IP, just the gateway

**Comment by bt** [comment]
I'm going to assume it has to do with @finndev's implementation of [Unknown6](https://github.com/keyphact/pgoapi/blob/master/pgoapi/protos/Signature.proto) which is now a `Signature` that is sent to the PGo servers describing your current environment.

Seeing how these properties were implemented would be interesting, and probably the reason why this is happening (as this data would be harder to fake):
```
    message SensorInfo {
        uint64 timestamp_snapshot = 1; // in ms
        double magnetometer_x = 3;
        double magnetometer_y = 4;
        double magnetometer_z = 5;
        double angle_normalized_x = 6;
        double angle_normalized_y = 7;
        double angle_normalized_z = 8;
        double accel_raw_x = 10;
        double accel_raw_y = 11;
        double accel_raw_z = 12;
        double gyroscope_raw_x = 13;
        double gyroscope_raw_y = 14;
        double gyroscope_raw_z = 15;
        double accel_normalized_x = 16;
        double accel_normalized_y = 17;
        double accel_normalized_z = 18;
        uint64 accelerometer_axes = 19; // Always 3
    }

    message DeviceInfo {
        string device_id = 1; // Hex string
        string android_board_name = 2;
        string android_bootloader = 3;
        string device_brand = 4; // On Android: product.brand
        string device_model = 5; // On Android: product.device
        string device_model_identifier = 6; // Android only, build.display.id
        string device_model_boot = 7;  // On Android: boot.hardware
        string hardware_manufacturer = 8; // On Android: product.manufacturer
        string hardware_model = 9;  // On Android: product.model
        string firmware_brand = 10; // On Android: product.name, on iOS: "iPhone OS"
        string firmware_tags = 12; // Android only, build.tags
        string firmware_type = 13; // On Android: build.type, on iOS instead: iOS version
        string firmware_fingerprint = 14; // Android only, build.fingerprint
    }
```

**Comment by curtiola** [comment]
Ahhh yeah I think you're onto something, maybe if they receive the same sensor information and such from the same IP it becomes a problem lol I really don't know much about this stuff though

**Comment by Thorben24** [comment]
@bt @curtiola @finndev 

With the IP ban im 100% sure, cause reset the ip adress fixed everything 2 times immediately. But in this case, maybe you will get an IP ban only if it detect its the same machine.

And if this source is true, it will be extremely important to send valid this informations to the pokego servers. A randomized feature would be great.

**Comment by Ray-PoGo-NL** [comment]
2 accounts signed in on the same machine results in a soft spell, it's obvious At That Point to Niantic
I tested it out I've really ip blocked access
